clients/v1_auth/v1_auth.go

```diff
@@ -0,0 +1,428 @@
+package v1_auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+	"github.com/influxdata/influx-cli/v2/pkg/influxid"
+)
+
+type Client struct {
+	clients.CLI
+	api.AuthorizationsApi
+	api.UsersApi
+	api.OrganizationsApi
+}
+
+type v1PrintOpts struct {
+	deleted bool
+	token   *v1Token
+	tokens  []v1Token
+}
+
+type v1Token struct {
+	ID          string   `json:"id"`
+	Description string   `json:"description"`
+	Token       string   `json:"token"`
+	Status      string   `json:"status"`
+	UserName    string   `json:"userName"`
+	UserID      string   `json:"userID"`
+	Permissions []string `json:"permissions"`
+}
+
+type CreateParams struct {
+	clients.OrgParams
+	Desc        string
+	Username    string
+	Password    string
+	NoPassword  bool
+	ReadBucket  []string
+	WriteBucket []string
+}
+
+func (c Client) Create(ctx context.Context, params *CreateParams) error {
+	if params.Password != "" && params.NoPassword {
+		return fmt.Errorf("only one of --password and --no-password may be specified")
+	}
+
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	ids, err := c.UsersApi.GetUsers(ctx).Name(params.Username).Execute()
+	if err != nil || len(ids.GetUsers()) == 0 {
+		return fmt.Errorf("could not get user from Username %q: %w", params.Username, err)
+	}
+
+	// verify an existing token with the same username doesn't already exist
+	auth, err := c.AuthorizationsApi.GetAuthorizationsID(ctx, ids.GetUsers()[0].GetId()).Execute()
+	authPtr := &auth
+	if &authPtr != nil {
+		return fmt.Errorf("authorization with username %q already exists", params.Username)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to verify username %q has no auth: %w", params.Username, err)
+	}
```

Comment:
I think these two checks are incompatible / need reworked. The query from line 55 is correct, but we want to see a 404 / empty array response to prove we aren't about to hit a name collision. The `GetAuthorizationsID` call based on ID shouldn't be needed.

You can use logic [like this](https://github.com/influxdata/influx-cli/blob/main/clients/backup/backup.go#L211-L212) to check if a non-`nil` `err` is a 404 response.